libs/tracing/unix/Tracing.ml

```diff
@@ -78,7 +79,7 @@ let configure_tracing service_name =
   (* This forwards the spans from Trace to the Opentelemetry collector *)
   Opentelemetry_trace.setup_with_otel_backend otel_backend
 
-let with_setup f =
+let with_setup f traces_endpoint =
```

Comment:
```suggestion
let with_setup traces_endpoint f =
```
The order of arguments matters! This is because we can curry functions (e.g. if we wanted to make a new function, `with_setup_dev_endpoint`, that always uses the dev endpoint, we could write `let with_setup_dev_endpoint = with_setup "semgrep-dev"`. Then we can run `with_setup_dev_endpoint f` and that'll be equivalent to writing `with_setup "semgrep-dev" f`). This makes code a lot nicer to read when we can take advantage of it. As a rule of thumb, configuration arguments should go in front.